test/Conversion/SCFToCalyx/convert_controlflow.mlir

```diff
@@ -0,0 +1,423 @@
+// CHECK-NEXT:           calyx.assign %std_add_0.left = %while_0_arg1_reg.out : i32
+// CHECK-NEXT:           calyx.assign %std_add_0.right = %while_0_arg2_reg.out : i32
+// CHECK-NEXT:           %0 = comb.and %bb3_arg0_reg.done, %bb3_arg1_reg.done : i1
+// CHECK-NEXT:           calyx.group_done %true, %0 ? : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @bb2_to_bb3  {
+// CHECK-NEXT:           calyx.assign %bb3_arg0_reg.in = %std_add_1.out : i32
+// CHECK-NEXT:           calyx.assign %bb3_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.assign %bb3_arg1_reg.in = %std_add_1.out : i32
+// CHECK-NEXT:           calyx.assign %bb3_arg1_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.assign %std_add_1.left = %while_0_arg1_reg.out : i32
+// CHECK-NEXT:           calyx.assign %std_add_1.right = %while_0_arg2_reg.out : i32
+// CHECK-NEXT:           %0 = comb.and %bb3_arg0_reg.done, %bb3_arg1_reg.done : i1
+// CHECK-NEXT:           calyx.group_done %true, %0 ? : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @assign_while_0_latch  {
+// CHECK-NEXT:           calyx.assign %while_0_arg0_reg.in = %std_add_2.out : i32
+// CHECK-NEXT:           calyx.assign %while_0_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.assign %while_0_arg1_reg.in = %bb3_arg0_reg.out : i32
+// CHECK-NEXT:           calyx.assign %while_0_arg1_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.assign %while_0_arg2_reg.in = %bb3_arg1_reg.out : i32
+// CHECK-NEXT:           calyx.assign %while_0_arg2_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.assign %std_add_2.left = %while_0_arg0_reg.out : i32
+// CHECK-NEXT:           calyx.assign %std_add_2.right = %in2 : i32
+// CHECK-NEXT:           %0 = comb.and %while_0_arg0_reg.done, %while_0_arg1_reg.done : i1
+// CHECK-NEXT:           %1 = comb.and %0, %while_0_arg2_reg.done : i1
+// CHECK-NEXT:           calyx.group_done %true, %1 ? : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @ret_assign_0  {
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.in = %while_0_arg2_reg.out : i32
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %ret_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.comb_group @comb_bb0_0  {
+// CHECK-NEXT:           calyx.assign %std_slt_0.left = %while_0_arg0_reg.out : i32
+// CHECK-NEXT:           calyx.assign %std_slt_0.right = %in1 : i32
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.comb_group @comb_bb0_1  {
+// CHECK-NEXT:           calyx.assign %std_slt_1.left = %while_0_arg1_reg.out : i32
+// CHECK-NEXT:           calyx.assign %std_slt_1.right = %while_0_arg2_reg.out : i32
+// CHECK-NEXT:         }
+// CHECK-NEXT:       }
+// CHECK-NEXT:       calyx.control  {
+// CHECK-NEXT:         calyx.seq  {
+// CHECK-NEXT:           calyx.enable @assign_while_0_init {compiledGroups = []}
```

Comment:
What is the `compiledGroups` attributed used for here?

Reply:
What is checked for here is the IR that is emitted from SCFToCalyx using the Calyx printers. This would indicate that the calyx printer for `calyx.enable` per default prints the `compiledGroups` attribute of `calyx.enable` https://github.com/llvm/circt/blob/072abc36c9a2347c50c8a86efb80f6f27405e8cd/include/circt/Dialect/Calyx/CalyxControl.td#L127-L133